.github/workflows/deploy-gh-pages.yml

```diff
@@ -15,13 +16,13 @@ jobs:
       - name: 🐍 Set up Python 3
         uses: actions/setup-python@v2
         with:
-          python-version: 3
+          python-version: '3.x'
```

Comment:
I recommend following [GitHub's docs](https://github.com/actions/setup-python) and use `'3.x'`, they tend to remove undocumented things with no notice.

For example, `pypy3` is being removed and you need to use something like `pypy-3.8` instead: https://github.com/actions/setup-python/issues/244.

Reply:
The docs you linked to refer to SemVer ranges -- whilst not linked, I believe it means `semver` as in the npm package, given it is used within the action (https://github.com/actions/setup-python/blob/156361d073e2d704ec0bfe646b4500cdc0547324/src/find-python.ts#L5). 

`semver` the package documents that the "x" is optional. (see https://github.com/npm/node-semver#x-ranges-12x-1x-12-)

> A partial version range is treated as an X-Range, so the special character is in fact optional.

Sorry, appreciate I didn't mention this originally.

A

Reply:
Taking a step back, I'm having trouble understanding what justifies spending a bunch of our time and churn researching, debating and reverting this instead of simply using what is in GitHub's official documentation and leaving it at that, rather than taking a risk just to fulfill one person's personal stylistic preference on a single key value. If I'd known it would become a huge deal I wouldn't have dreamed of touching it, and I certainly won't make that mistake again—and I should have known better, from past experience.